### PR TITLE
Add documentation for beforeMarshal event.

### PR DESCRIPTION
**Pull request by markstory**
Related to cakephp/cakephp#5704

### CODE REVIEW

**Comment by antograssiot** [comment]
@markstory I pushed my comment as a commit

File: en/orm/saving-data.rst [hunk 1]
```diff
@@ -279,6 +279,25 @@ array::
...
+
+Modifying Request Data Before Building Entities
+-----------------------------------------------
+
+If you need to modify request data before it is converted into entities, you can
+use the ``Model.beforeMarshal`` event. This event lets you manipulate the
+request data just before entities are created::
+
+    // In a table or behavior class
+    public function beforeMarshal(Event $event, $data, $options = [])
```
Comment (dereuromark) [inline en/orm/saving-data.rst]: Type hinting is off here.

File: en/orm/table-objects.rst [hunk 2]
```diff
@@ -123,6 +123,14 @@ To add an event listener to a Table class or Behavior simply implement the
 method signatures as described below. See the :doc:`/core-libraries/events` for
 more detail on how to use the events subsystem.
 
+beforeMarshal
+-------------
+
+.. php:method:: beforeMarshal(Event $event, ArrayObject $data, $options)
```
Comment (dereuromark) [inline en/orm/table-objects.rst]: Why the type hint on the $data and not $options (as for all other events?)

Comment (markstory) [inline en/orm/table-objects.rst]: Because I am lazy :)

**Pull request merged**